src/coord/src/coord.rs

```diff
@@ -992,6 +1006,45 @@ where
         )
     }
 
+    /// Inserts a single row into the MZ_COLUMNS catalog view, with diff `diff`.
+    /// The MZ_COLUMNS catalog view contains a row for each column in every Table, Source, and View.
+    ///
+    /// # Arguments
+    ///
+    /// * `desc`:        RelationDesc of the Table, Source, or View.
+    /// * `full_name`:   Full, qualified name of the Table, Source, or View (e.g., "materialize.public.table").
+    /// * `global_id`:   GlobalId of the Table, Source, or View.
+    /// * `diff`:        The number of updates to perform. A positive number indicates an addition
+    ///                  of the row, a negative number indicates a subtraction.
+    fn update_mz_columns_catalog_view(
+        &mut self,
+        desc: &RelationDesc,
+        full_name: &str,
+        global_id: GlobalId,
+        diff: isize,
+    ) {
+        for (i, (column_name, column_type)) in desc.iter().enumerate() {
+            self.update_catalog_view(
+                MZ_COLUMNS.id,
+                iter::once((
+                    Row::pack(&[
+                        Datum::String(&full_name),
+                        Datum::String(&global_id.to_string()),
+                        Datum::String(&i.to_string()),
+                        Datum::String(
+                            &column_name
+                                .map(|n| n.to_string())
+                                .unwrap_or_else(|| "?".to_owned()),
```

Comment:
If a test enforces the use of `?` here don't rock the boat, but if not `?column?` is the standard "missing column" identifier to use.